sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptionDataV1.java

```diff
@@ -0,0 +1,192 @@
+     *         key bytes.
+     */
+    @Override
+    public WrappedKey getWrappedContentKey() {
+        return this.wrappedContentKey;
+    }
+
+    /**
+     * Gets the encryption agent that is used to identify the encryption protocol version and encryption algorithm.
+     *
+     * @return an {@link EncryptionAgent}.
+     */
+    @Override
+    public EncryptionAgent getEncryptionAgent() {
+        return this.encryptionAgent;
+    }
+
+    /**
+     * Gets the content encryption IV.
+     *
+     * @return The content encryption IV.
+     */
+    byte[] getContentEncryptionIV() {
+        return this.contentEncryptionIV;
+    }
+
+    /**
+     * Gets the metadata for encryption.
+     *
+     * @return A HashMap containing the encryption metadata in a key-value format.
+     */
+    @Override
+    public Map<String, String> getKeyWrappingMetadata() {
+        return this.keyWrappingMetadata;
+    }
+
+    /**
+     * Sets the encryption mode
+     *
+     * @param encryptionMode The encryption mode
+     *
+     * @return this
+     */
+    EncryptionDataV1 setEncryptionMode(String encryptionMode) {
+        this.encryptionMode = encryptionMode;
+        return this;
+    }
+
+    /**
+     * Sets the wrapped key that is used to store the wrapping algorithm, key identifier and the encrypted key bytes.
+     *
+     * @param wrappedContentKey A {@link WrappedKey} object that stores the wrapping algorithm, key identifier and the
+     *         encrypted key bytes.
+     *
+     * @return this
+     */
+    EncryptionDataV1 setWrappedContentKey(WrappedKey wrappedContentKey) {
+        this.wrappedContentKey = wrappedContentKey;
+        return this;
+    }
+
+    /**
+     * Sets the encryption agent that is used to identify the encryption protocol version and encryption algorithm.
+     *
+     * @param encryptionAgent The {@link EncryptionAgent}.
+     *
+     * @return this
+     */
+    EncryptionDataV1 setEncryptionAgent(EncryptionAgent encryptionAgent) {
+        this.encryptionAgent = encryptionAgent;
+        return this;
+    }
+
+    /**
+     * Sets the content encryption IV.
+     *
+     * @param contentEncryptionIV The content encryption IV.
+     *
+     * @return this
+     */
+    EncryptionDataV1 setContentEncryptionIV(byte[] contentEncryptionIV) {
+        this.contentEncryptionIV = contentEncryptionIV;
+        return this;
+    }
+
+    /**
+     * Sets the metadata for encryption.
+     *
+     * @param keyWrappingMetadata A HashMap containing the encryption metadata in a key-value format.
+     *
+     * @return this
+     */
+    EncryptionDataV1 setKeyWrappingMetadata(Map<String, String> keyWrappingMetadata) {
+        this.keyWrappingMetadata = keyWrappingMetadata;
+        return this;
+    }
+
+    @Override
+    public String toJsonString() throws JsonProcessingException {
+        return MAPPER.writeValueAsString(this);
```

Comment:
we should use object serializer from core.

Reply:
I'll switch all the models in this package to use that.